sdks/python/apache_beam/io/gcp/gcsfilesystem.py

```diff
@@ -352,27 +352,20 @@ def delete(self, paths):
     Args:
```

Comment:
Is line 350 comment still relevant? The directories are not getting deleted recursively.

Reply:
GCS is currently a flat system. I think it will match and delete all the files with that prefix, which is equivalent to delete the files recursively.

Reply:
https://github.com/apache/beam/pull/29477/files - this follows a pattern to local filesystem. Would this work?

Reply:
Have you debugged the original code? I am curious of which part doesn't work. I guess it is the file matching part?

Reply:
Yes, the matching part. When a directory is provided, we append `*`. 

* expected - all the dirs and objects should be deleted.
* Actual - only objects are getting deleted.

Solution - let's use `bucket.list_buckets()` and match the path with the provided dirs and delete them. I can spin up a PR if this sounds good.

Reply:
That makes sense to me. Thanks for working on this! Please go ahead to create a PR.